internal/mode/static/state/graph/gatewayclass.go

```diff
@@ -75,11 +76,27 @@ func buildGatewayClass(gc *v1beta1.GatewayClass) *GatewayClass {
 	}
 }
 
-func validateGatewayClass(gc *v1beta1.GatewayClass) error {
+func validateGatewayClass(gc *v1beta1.GatewayClass, npCfg *ngfAPI.NginxProxy) error {
 	if gc.Spec.ParametersRef != nil {
-		path := field.NewPath("spec").Child("parametersRef")
-		return field.Forbidden(path, "parametersRef is not supported")
+		if _, ok := supportedParamKinds[string(gc.Spec.ParametersRef.Kind)]; !ok {
+			path := field.NewPath("spec").Child("parametersRef")
+			return field.Forbidden(path, "parametersRef resource not allowed")
+		}
+
+		if gc.Spec.ParametersRef.Namespace == nil {
+			path := field.NewPath("spec").Child("parametersRef")
+			return field.Required(path, "parametersRef namespace must be specified for NginxProxy")
+		}
+
+		if npCfg == nil {
+			path := field.NewPath("spec").Child("parametersRef")
+			return field.NotFound(path, "parametersRef resource not found")
```

Comment:
can we use field.Invalid here instead? Similar to https://github.com/nginxinc/nginx-gateway-fabric/blob/09fe18c212ecaaf11f6b114cc16739cdb1f0f15c/internal/mode/static/state/graph/gateway_listener.go#L445
note that field.NotFound accepts field value not an error message

Reply:
Wouldn't the error message be redundant? NotFound is pretty clear what went wrong. I can fix the message to be the field name.

Invalid is intended for malformed values.